utils/errors/errors.go

```diff
@@ -0,0 +1,114 @@
+package errors
+
+import (
+	"context"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/gogo/protobuf/grpc"
+	"github.com/tendermint/tendermint/libs/log"
+	"golang.org/x/xerrors"
+	grpc2 "google.golang.org/grpc"
+)
+
+type causer interface {
+	Cause() error
+}
+
+// Is checks if the error is a (wrapped) error of the given type
+func Is[T error](err error) bool {
+	switch e := err.(type) {
+	case nil:
+		return false
+	case T:
+		return true
+	case causer:
+		return Is[T](e.Cause())
+	case xerrors.Wrapper:
+		return Is[T](e.Unwrap())
+	default:
+		return false
+	}
+}
+
+// ServerWithSDKErrors wraps around a grpc server to return registered errors
+type ServerWithSDKErrors struct {
```

Comment:
Having this defined in `errors` package feels weird to me.

Reply:
agree, but it's been there before, I just gave it a more meaningful name. Any suggestions where to put it?

Reply:
This to me is very grpc-related. Consider creating a new `grpc` package in utils?